From 480bc71deb5687cc7e9790343849707bb73c889b Mon Sep 17 00:00:00 2001
From: Christine Gharzuzi <cgharzuzi@habana.ai>
Date: Thu, 30 Apr 2020 14:54:08 +0300
Subject: [PATCH] [SW-0] pci: training: fix gen3 and gen4 PCI training issue

Fix PCIe Gen3 and Gen4 Training issue that is causing
enumeration failures, link down or unassigned memory.

fix training by enabling P4-P9 only, avoiding initial preset
and enable 2ms timeout + 24ms timeout(23 exit).

Change-Id: I9be7825f48355126bcfdbc4716233139bd1a4b2a
Signed-off-by: Christine Gharzuzi <cgharzuzi@habana.ai>
---

diff --git a/gaudi/common.c b/gaudi/common.c
index 585ab60..d69ac42 100644
--- a/gaudi/common.c
+++ b/gaudi/common.c
@@ -75,8 +75,13 @@
 	regmap_writel(0x10001DA3,
 		      mmPCIE_DBI_SUBSYSTEM_ID_SUBSYSTEM_VENDOR_ID_REG);
 
-	/* Disable the EQ REDO */
+	/* Disable the EQ REDO Gen3 */
 	regmap_writel(0x00402801, mmPCIE_DBI_GEN3_RELATED_OFF);
+	regmap_writel(0x04027041, mmPCIE_DBI_GEN3_EQ_CONTROL_OFF);
+
+	/* Disable the EQ REDO Gen4*/
+	regmap_writel(0x01402801, mmPCIE_DBI_GEN3_RELATED_OFF);
+	regmap_writel(0x04027041, mmPCIE_DBI_GEN3_EQ_CONTROL_OFF);
 
 	/* delay before executing Synopsys W/A > 300us */
 	udelay(576);
